### PR TITLE
Add dependabot config

### PR DESCRIPTION
**Pull request by plkokanov**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
Adds `dependabot.yaml` config for automatic upgrades of `gardener/gardener` and `golang` versions.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: .github/workflows/vendor_gardener.yaml
Comment (ialidzhikov) [inline .github/workflows/vendor_gardener.yaml:1]: In https://github.com/gardener/gardener-extension-registry-cache/pull/147/commits/f683d23a862bd1473ebc30ddf118213247ccfb99, I dropped this Github action with the vendor dir removal. I think the only use of this Github action after the vendor dir removal would be to update the 2 script that are currently coming from gardener (component_descriptor and prepare_release). I decided to manually maintain these 2 scripts instead of having such Github action.

No strong opinion, I just wanted to share the approach I decided to follow for the registry-cache repo.

Comment (plkokanov) [reply .github/workflows/vendor_gardener.yaml:1]: Yep I noticed that, but since I sometimes tend to forget such stuff, I decided to leave the action :)

Comment (ialidzhikov) [inline .github/workflows/vendor_gardener.yaml:1]: Maybe you could rename the file and the name field to something other than vendoring? (as we no longer have the vendor dir or make target named vendor)

Comment (plkokanov) [reply .github/workflows/vendor_gardener.yaml:1]: wdyt about something like "Sync Gardener Script Dependencies"

Comment (ialidzhikov) [reply .github/workflows/vendor_gardener.yaml:1]: No strong opinion. The above suggestion is ok. Alternatively, you can use something around make tidy as at the end the github action does this. Whatever you prefer.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Comment by ialidzhikov** [comment]
/hold
to unblock tide - this PR has changes under .github and tide cannot merge such PRs

**Pull request merged**
